blockchain_connector/Dockerfile

```diff
@@ -64,35 +86,15 @@ RUN OPENSSL_VER=1.1.1d \
  && make install -j$THREADS
 
 
-# Build Avalon blockchain connector intermediate docker image
-FROM base_image as build_image
-# Add necessary packages to build
-RUN apt-get update \
- && apt-get install -y -q \
-    pkg-config \
-    protobuf-compiler \
-    libprotobuf-dev \
-    software-properties-common \
-    dh-autoreconf \
-    ocaml \
-    xxd \
-    ocamlbuild \
-    cmake \
-    make \
-    git \
-    swig \
- && apt-get -y -q upgrade \
- && apt-get clean \
- && rm -rf /var/lib/apt/lists/*
-
 
-# Installing web3 for ethereum
-RUN pip3 install --upgrade setuptools json-rpc web3 py-solc-x nose2 eth-abi
+# -------------=== common/python build ===-------------
 
-# Install specific version of solc to support versions <0.6.0
-RUN python3 -m solcx.install v0.5.15
+#Build common/cpp intermediate docker image
+FROM python_image as common_python_image
 
-RUN pip3 install nest_asyncio
+RUN apt-get update \
+ && apt-get install -y -q \
+    make
```

Comment:
make also can be installed in python_image. This will avoid below steps which is repetitive in each image. Can we rename python_image as base_python_image.
```
RUN apt-get update \
 && apt-get install -y -q \
    make
```

Reply:
common_python_image is for this purpose. likewise, common_crypto_utils_image is for common/crypto_utils, avalon_sdk_image is for sdk/. These images will be created once and every other modules will reuse the cached images.

Reply:
If `make` is installed as part of base `python_image` then we don;t need to install `make` again in common_python_image, common_crypto_utils and avalon_sdk_image.

Reply:
This can be addressed in seperate PR.

Reply:
I thought it's just a one line change. Did you see any issue in adding `make` in python_image. Is this breaking something ?

Reply:
I misunderstood it! python_image is a kind of base image for python package. common_python_package, common_crypto_utils_image, avalon_sdk_image use python_image as their base image. and this happens across different avalon components.

Reply:
So installing `make` as part of python_image would help instead of installing `make'  in each of common_python_package, common_crypto_utils_image, avalon_sdk_image  ?

Reply:
Done